### ISSUE TITLE
Generate pytest test suite for all modules in repo

### ISSUE THREAD

**Issue by abuzarmahmood** [abuzarmahmood/blechRNN#22]

**Comment by abuzarmahmood** [comment]
### Summary of Suggestions and Changes

1. **Identification of Files for Test Suite:**
   - The files identified for modification to generate a pytest test suite are:
     - `/home/exouser/Desktop/blech_github_bot/repos/abuzarmahmood/blechRNN/src/model.py`: Contains core model definitions and logic for the recurrent neural network.
     - `/home/exouser/Desktop/blech_github_bot/repos/abuzarmahmood/blechRNN/src/run_model.py`: Responsible for running the model on data and generating outputs.
     - `/home/exouser/Desktop/blech_github_bot/repos/abuzarmahmood/blechRNN/src/train.py`: Handles model training and prediction.

2. **Plan to Address the Issue:**
   - Create test cases for the functions and classes defined in the identified files.
   - Ensure correctness of outputs, handle edge cases, and verify expected behavior under various conditions.

3. **Specific Changes and Code Snippets:**

   - **Test Suite for `model.py`:**
     - **File:** `tests/test_model.py`
     - **Description:** Verify functionality of classes `CTRNN`, `CTRNN_plus_output`, and `autoencoderRNN`.
     - **Code Snippet:**
       ```python
       import pytest
       import torch
       from src.model import CTRNN, CTRNN_plus_output, autoencoderRNN

       def test_ctrnn_initialization():
           model = CTRNN(input_size=10, hidden_size=20)
           assert model.input_size == 10
           assert model.hidden_size == 20

       def test_ctrnn_forward():
           model = CTRNN(input_size=10, hidden_size=20)
           input_data = torch.randn(5, 3, 10)  # seq_len, batch, input_size
           output, hidden = model(input_data)
           assert output.shape == (5, 3, 20)
           assert hidden.shape == (3, 20)

       def test_autoencoder_rnn_forward():
           model = autoencoderRNN(input_size=10, hidden_size=20, output_size=5)
           input_data = torch.randn(5, 3, 10)  # seq_len, batch, input_size
           output, latent_out = model(input_data)
           assert output.shape == (5, 3, 5)
           assert latent_out.shape == (5, 3, 20)
       ```

   - **Test Suite for `run_model.py`:**
     - **File:** `tests/test_run_model.py`
     - **Description:** Verify data processing and model execution logic.
     - **Code Snippet:**
       ```python
       import pytest
       import numpy as np
       from src.run_model import process_data

       def test_data_processing():
           # Assuming process_data is a function that processes input data
           input_data = np.random.rand(100, 10)
           processed_data = process_data(input_data)
           assert processed_data.shape == (100, 10)  # Expected shape after processing
       ```

   - **Test Suite for `train.py`:**
     - **File:** `tests/test_train.py`
     - **Description:** Verify training logic and loss functions.
     - **Code Snippet:**
       ```python
       import pytest
       import torch
       from src.train import train_model, smooth_MSELoss

       def test_train_model():
           # Mock model, inputs, and labels for testing
           model = torch.nn.Linear(10, 5)
           inputs = torch.randn(5, 3, 10)
           labels = torch.randn(5, 3, 5)
           trained_model, loss_history, cross_val_loss = train_model(
               model, inputs, labels, output_size=5, train_steps=10
           )
           assert len(loss_history) > 0

       def test_smooth_mse_loss():
           criterion = smooth_MSELoss()
           input_data = torch.randn(5, 3, 10)
           target_data = torch.randn(5, 3, 10)
           loss = criterion(input_data, target_data)
           assert loss.item() >= 0
       ```

These test suites are designed to ensure that the core functionalities of the modules are tested and verified. Additional specific cases and edge conditions can be added as needed.

---
*This response was automatically generated by blech_bot using model gpt-4o*

**Comment by abuzarmahmood** [comment]
[ develop_issue ]

**Comment by abuzarmahmood** [comment]
Created pull request: https://github.com/abuzarmahmood/blechRNN/pull/24
Continue discussion there.

---
*This response was automatically generated by blech_bot using model gpt-4o*
